pkg/await/await.go

```diff
@@ -104,10 +105,13 @@ func Read(
 		return nil, err
 	}
 
-	// No inputs means that we do not manage the resource, i.e., it's a call to
-	// `CustomResource#get`. Simply return the object.
-	if inputs == nil || len(inputs.Object) == 0 {
-		return clientForResource.Get(name, metav1.GetOptions{})
+	outputs, err := clientForResource.Get(name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	} else if inputs == nil || len(inputs.Object) == 0 {
+		// No inputs means that we do not manage the resource, i.e., it's a call to
+		// `CustomResource#get`. Simply return the object.
+		return outputs, nil
```

Comment:
Not sure I follow what's motivating the above change - what issue is this fixing?

Reply:
Recapping offline discussion, in the case of `.get` we do _not_ want to run validation routines  -- we want to return the object itself, in whatever stat it is.

In the case of a `refresh`, we want to validate the outputs are in a valid state.

We run `clientForResource.Get` so that we can pass `ouptuts` to the validation logic below.